src/rust/engine/process_execution/src/local.rs

```diff
@@ -360,7 +362,10 @@ pub trait CapturedWorkdir {
       // code. The idea going forward though is we eventually want to pass incremental results on
       // down the line for streaming process results to console logs, etc. as tracked by:
       //   https://github.com/pantsbuild/pants/issues/6089
-      .and_then(ChildResults::collect_from)
+      .map(ChildResults::collect_from)
```

Comment:
Can maybe skip the `map` call and move the `ChildResults::collect_from(stream)` into your `and_then`?

Reply:
BUT, it's green. So could also ignore this and merge: just a nit.